Client/src/Components/Graph/GraphControls/AxesControl.tsx

```diff
@@ -29,48 +31,80 @@ const buildVariableList = (datasets: IDatasetModel[]): IVariable[] => {
 }
 
 export const AxesControl = (props: IProps) => {
-  const { datasets, axes, onAxesChange } = { ...props }
+  const { datasets, axes, onAxesChange, dimensions } = { ...props }
   const classes = classStyles()
 
-  //todo unhardcode the variables
   const [showSettings, setSettingsToggle] = useState(true)
 
   const [xVariableMissing, setXVariableMissing] = useState([])
   const [yVariableMissing, setYVariableMissing] = useState([])
-  const [xUnits, setXUnits] = useState([])
-  const [yUnits, setYUnits] = useState([])
+  const [xUnits, setXUnits] = useState<IUnitModel[]>([])
+  const [yUnits, setYUnits] = useState<IUnitModel[]>([])
   const [variables, setVariables] = useState<IVariable[]>([])
 
   useEffect(() => {
     setVariables(buildVariableList(datasets))
   }, [datasets])
 
+  const getVariableDimensions = (datasets: IDatasetModel[], variableName): number => {
+
+    const dictionary = {}
+    const matchingDatasetID = []
+    const dimensionID = []
+
+    for (var i = 0; i < datasets.length; i++) {
+      for (var j = 0; j < datasets[i].data.variables.length; j++) {
+        if (datasets[i].data.variables[j].name == variableName) {
+
+          dimensionID.push(datasets[i].data.variables[j].dimensionId)
+          matchingDatasetID.push(datasets[i].id)
+          dictionary[datasets[i].data.variables[j].dimensionId] = matchingDatasetID
```

Comment:
you're not resetting matchingDatasetId, so if the dimension doesnt match it will contain all the other dataset ids

Reply:
refactor the double loop into this     
datasets.forEach(dataset => {
      const foundVariable = dataset.data.variables.find(variable => variable.name == variableName)
      ...
    })

Reply:
this should work:     datasets.forEach(dataset => {
      const foundVariable = dataset.data.variables.find(variable => variable.name == variableName)
      const datasetIds = dictionary[foundVariable.dimensionId] || []
      datasetIds.push(dataset.id)
    })

remove dimensionId

Reply:
ill just push a commit, Ill add some tests too

Reply:
i can fix that @ds161833

Reply:
Gonna push a change to fix this.